android-lib/src/main/java/com/kinvey/android/cache/ClassHash.java

```diff
@@ -860,8 +860,13 @@ private static int deleteClassData(String collection, DynamicRealm realm, Class<
                             ret.put(info.getName(), c);
                         }
                     }
-                } else {
-                    ret.put(info.getName(), o);
+                } else { //if you are here, then the field is a primitive
+                    // realm keeps int values as a long, so we need to convert it back
+                    if (info.getType().isPrimitive() && info.getType().getSimpleName().equals("int") && o instanceof Long) {
+                        ret.put(info.getName(), ((Long) o).intValue());
```

Comment:
Converting `long` to `int` could result in losing/truncating data. Is this guaranteed to be a `int` in this case?

Reply:
Yes, This is be a `int` because we check that this field is `int` before converting`info.getType().getSimpleName().equals("int")`